OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -81,14 +81,17 @@ public virtual void Tick(Actor self)
 			else
 				realignTick = 0;
 
-			var df = DesiredFacing ?? (facing != null ? facing.Facing : TurretFacing);
-			TurretFacing = Util.TickFacing(TurretFacing, df, info.ROT);
+			self.World.AddFrameEndTask(w =>
```

Comment:
Instead of making a new function that adds a bit of state to check if the turret has rotated yet this tick, I can just check with Utils.TickFacing where needed and rotate it at the end of the frame. There would be a significant amount of plumbing and mess involved in making sure that the turret doesn't rotate twice per tick if I were to remove the FrameEndTask.